libs/build_pio.sh

```diff
@@ -53,53 +53,38 @@ else
   branch=pio$(echo $version | sed -e 's/\./_/g')
 fi
 
-URL="https://github.com/NCAR/ParallelIO/releases/download/$branch/${software}.tar.gz"
-[[ -f $software.tar.gz ]] || $WGET $URL
-
-tar -xf ${software}.tar.gz
+URL=https://github.com/NCAR/ParallelIO
+[[ -d $software ]] || git clone $URL $software
+[[ -d $software ]] && cd $software || ( echo "$software does not exist, ABORT!"; exit 1 )
 
+git checkout $branch
 
-[[ -d $software ]] && cd $software || ( echo "$software does not exist, ABORT!"; exit 1 )
+# These repositories are used internally by PIO. Download them so DOWNLOAD_ONLY option works.
+[[ -d CMake_Fortran_utils ]] || git clone https://github.com/CESM-Development/CMake_Fortran_utils
+[[ -d genf90 ]] || git clone https://github.com/PARALLELIO/genf90.git
+CMAKE_FLAGS="-DUSER_CMAKE_MODULE_PATH=`pwd`/CMake_Fortran_utils -DGENF90_PATH=`pwd`/genf90"
```

Comment:
not that I care whatever PIO is doing, but are these "required" by PIO?
If they are required, does anyone know why these are not git submodules?
If they are not required, what do we need them for?

Reply:
if these are required for testing, better to put them in an if-block:
```
if [[ $MAKE_CHECK =~ [yYtT] ]]; then
...
fi
```

Reply:
Yes, they are required, if you want the ability to use `DOWNLOAD_ONLY`. Otherwise, they are downloaded by CMake at build time.